deps/verifier/src/tpm/mod.rs

```diff
@@ -0,0 +1,439 @@
+    fn to_quote(&self) -> Result<Quote> {
+        // Extract raw signature bytes from the marshalled TSS signature
+        // Guest-components marshals the entire Signature struct, but az_cvm_vtpm expects raw bytes
+        let raw_signature_bytes = self.extract_raw_signature_bytes()?;
+
+        // Decode the attestation message
+        let message = general_purpose::STANDARD
+            .decode(&self.message)
+            .context("Failed to decode message from base64")?;
+
+        // Parse PCR values from hex strings to byte arrays
+        let pcrs = self.parse_pcr_values()?;
+
+        // Construct Quote with the corrected signature format
+        let quote_data = serde_json::json!({
+            "signature": raw_signature_bytes,
+            "message": message,
+            "pcrs": pcrs
+        });
+
+        serde_json::from_value::<Quote>(quote_data)
+            .context("Failed to construct Quote from parsed data")
+    }
+
+    fn extract_raw_signature_bytes(&self) -> Result<Vec<u8>> {
+        let signature_marshalled = general_purpose::STANDARD
+            .decode(&self.signature)
+            .context("Failed to decode signature from base64")?;
+
+        let tss_signature = Signature::unmarshall(&signature_marshalled)
+            .context("Failed to unmarshal TSS signature")?;
+
+        match tss_signature {
+            Signature::RsaSsa(rsa_sig) => Ok(rsa_sig.signature().to_vec()),
+            _ => bail!("Unsupported signature type, expected RSA-SSA"),
+        }
+    }
+
+    fn parse_pcr_values(&self) -> Result<Vec<[u8; 32]>> {
+        let mut pcrs = Vec::new();
+        for pcr_str in &self.pcrs {
+            if pcr_str.len() != 64 {
+                bail!(
+                    "PCR should be 64 hex characters (32 bytes), got {}",
+                    pcr_str.len()
+                );
+            }
+            let pcr_bytes = decode_hex(pcr_str)
+                .with_context(|| format!("Failed to decode PCR from hex: {}", pcr_str))?;
+            if pcr_bytes.len() != 32 {
+                bail!("PCR should be exactly 32 bytes, got {}", pcr_bytes.len());
+            }
+            let pcr_array: [u8; 32] = pcr_bytes.try_into()
+                .map_err(|_| anyhow!("PCR must be exactly 32 bytes"))?;
+            pcrs.push(pcr_array);
+        }
+        Ok(pcrs)
+    }
+}
+
+#[derive(Debug)]
+pub struct TpmVerifier {
+    trusted_ak_hashes: HashSet<Vec<u8>>,
+}
+
+impl Default for TpmVerifier {
+    fn default() -> Self {
+        let config = config::TpmVerifierConfig::default();
+        Self::new(config).unwrap_or_else(|_| Self {
+            trusted_ak_hashes: HashSet::new(),
+        })
+    }
+}
+
+impl TpmVerifier {
+    pub fn new(config: config::TpmVerifierConfig) -> Result<Self> {
+        let mut trusted_ak_hashes = HashSet::new();
+        if let Some(keys_dir) = config.trusted_ak_keys_dir {
+            let entries = fs::read_dir(&keys_dir)?.collect::<Result<Vec<_>, _>>()?;
```

Comment:
I am not sure if `.` and `..` will be treated as an entry. If so, we might need to skip them